opslevel/datasource_opslevel_service.go

```diff
@@ -1,210 +1,231 @@
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure ServiceDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &ServiceDataSource{}
+
+func NewServiceDataSource() datasource.DataSource {
+	return &ServiceDataSource{}
+}
+
+// ServiceDataSource manages a Service data source.
+type ServiceDataSource struct {
+	CommonDataSourceClient
+}
+
+// ServiceDataSourceModel describes the data source data model.
+type ServiceDataSourceModel struct {
+	Alias                      types.String `tfsdk:"alias"`
+	Aliases                    types.List   `tfsdk:"aliases"`
+	ApiDocumentPath            types.String `tfsdk:"api_document_path"`
+	Description                types.String `tfsdk:"description"`
+	Framework                  types.String `tfsdk:"framework"`
+	Id                         types.String `tfsdk:"id"`
+	Language                   types.String `tfsdk:"language"`
+	LifecycleAlias             types.String `tfsdk:"lifecycle_alias"`
+	Name                       types.String `tfsdk:"name"`
+	Owner                      types.String `tfsdk:"owner"`
+	OwnerId                    types.String `tfsdk:"owner_id"`
+	PreferredApiDocumentSource types.String `tfsdk:"preferred_api_document_source"`
+	Product                    types.String `tfsdk:"product"`
+	Properties                 types.List   `tfsdk:"properties"`
+	Repositories               types.List   `tfsdk:"repositories"`
+	Tags                       types.List   `tfsdk:"tags"`
+	TierAlias                  types.String `tfsdk:"tier_alias"`
+}
+
+func NewServiceDataSourceModel(ctx context.Context, service opslevel.Service) (ServiceDataSourceModel, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	serviceDataSourceModel := ServiceDataSourceModel{
+		ApiDocumentPath: types.StringValue(service.ApiDocumentPath),
+		Description:     types.StringValue(service.Description),
+		Id:              types.StringValue(string(service.Id)),
+		Framework:       types.StringValue(service.Framework),
+		Language:        types.StringValue(service.Language),
+		LifecycleAlias:  types.StringValue(service.Lifecycle.Alias),
+		Name:            types.StringValue(service.Name),
+		Owner:           types.StringValue(service.Owner.Alias),
+		OwnerId:         types.StringValue(string(service.Owner.Id)),
+		Product:         types.StringValue(service.Product),
+		TierAlias:       types.StringValue(service.Tier.Alias),
+	}
+
+	serviceAliases, svcDiags := types.ListValueFrom(ctx, types.StringType, service.Aliases)
+	diags = append(diags, svcDiags...)
+	serviceDataSourceModel.Aliases = serviceAliases
+
+	if service.PreferredApiDocumentSource == nil {
+		serviceDataSourceModel.PreferredApiDocumentSource = types.StringNull()
```

Comment:
Likewise with `ListNull()`, etc.

Reply:
We don't really need `StringNull()` here - using `serviceDataSourceModel.PreferredApiDocumentSource = types.StringNull()` is the same as leaving the `serviceDataSourceModel.PreferredApiDocumentSource` field unset (null). Terraform is able to infer the type here.

However, a few lines down we do need to be explicit with `types.ListNull(types.StringType)` since Terraform can't be sure what kind of a null list to work with. We end up with something like:
```tf
│ Expected framework type from provider logic: types.ListType[basetypes.StringType] / underlying type: tftypes.List[tftypes.String]
│ Received framework type from provider logic: types.ListType[!!! MISSING TYPE !!!] / underlying type: tftypes.List[tftypes.DynamicPseudoType]
```